platform/android/src/java/lang.hpp

```diff
@@ -1,19 +1,63 @@
 #pragma once
 
+#include "../jni/binding.hpp"
+
 namespace mbgl {
 namespace android {
 namespace java {
 namespace lang {
 
-class Float {
-public:
-    static constexpr auto Name() { return "java/lang/Float"; };
-};
+using ObjectTag = jni::ObjectTag; // == "java/lang/Object"
+using Object = jni::Object<>; // == jni::Object<jni::ObjectTag>
+
+using StringTag = jni::StringTag; // == "java/lang/String"
+using String = jni::String; // == jni::Object<jni::StringTag>
+
+struct ObjectArrayTag { static constexpr auto Name() { return "[Ljava/lang/Object;"; } };
+using ObjectArray = jni::Object<ObjectArrayTag>;
+
+struct BooleanTag { static constexpr auto Name() { return "java/lang/Boolean"; } };
+using Boolean = jni::Object<BooleanTag>;
+using Boolean_constructor = binding::Constructor<BooleanTag, jni::jboolean>;
+struct Boolean_booleanValueTag { static constexpr auto Name() { return "booleanValue"; } };
+using Boolean_booleanValue = binding::Method<BooleanTag, Boolean_booleanValueTag, jni::jboolean(void)>;
```

Comment:
It's not necessary, but I added it for clarity and to disambiguate it from a function call. We can remove it if you prefer.